cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -173,6 +174,12 @@ func (kc *kubeletConfig) Default(cfg *kubeadmapi.ClusterConfiguration, _ *kubead
 		warnDefaultComponentConfigValue(kind, "healthzBindAddress", kubeletHealthzBindAddress, kc.config.HealthzBindAddress)
 	}
 
+	if kc.config.RootDir == "" {
+		kc.config.RootDir = kubeletconfigv1beta1.DefaultRootDir
+	} else if kc.config.RootDir != kubeletconfigv1beta1.DefaultRootDir {
+		warnDefaultComponentConfigValue(kind, "rootDir", kubeletconfigv1beta1.DefaultRootDir, kc.config.RootDir)
+	}
```

Comment:
you can omit this change from kubeadm given there is no need to override the default or print a warning if the user diverted from the default.

also, kubeletconfigv1beta1 imports the internal kubelet type which is not desired.